candy-machine/program/src/processor/mint.rs

```diff
@@ -106,6 +106,16 @@ pub fn handle_mint_nft<'info>(
     let instruction_sysvar_account_info = instruction_sysvar_account.to_account_info();
     let instruction_sysvar = instruction_sysvar_account_info.data.borrow();
     let current_ix = get_instruction_relative(0, &instruction_sysvar_account_info).unwrap();
+    if candy_machine.items_redeemed >= candy_machine.data.items_available {
+        punish_bots(
+            error!(CandyError::CandyMachineEmpty).into(),
```

Comment:
i had to make some changes to `punish_bots`. this can now just be changed back to just `CandyError::CandyMachineEmpty`